instrumentation/CONTRIBUTING.md

```diff
@@ -0,0 +1,483 @@
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation/werewolf.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation/werewolf/instrumentation.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation/werewolf/version.rb
+      create  instrumentation/werewolf/test/test_helper.rb
+      create  instrumentation/werewolf/test/opentelemetry/instrumentation/werewolf/instrumentation_test.rb
+      insert  .toys/.data/releases.yml
+      insert  instrumentation/all/Gemfile
+      insert  instrumentation/all/opentelemetry-instrumentation-all.gemspec
+      insert  instrumentation/all/lib/opentelemetry/instrumentation/all.rb
+
+```
+
+## Implementation guidelines
+
+The original design and implementation of this project was heavily influenced by Datadog's `dd-trace-rb` project. You may refer to the [Datadog Porting Guide](datadog-porting-guide.md) as a reference for implementing instrumentations, however the following guidelines are specific to OpenTelemetry Ruby:
+
+* Use `OpenTelemetry::Instrumentation::Base`
+* Use the OpenTelemetry API
+* Use First Party Extension Points
+* Use Semantic Conventions
+* Write comprehensive automated tests
+* Understand Performance Characteristics
+
+### Use `OpenTelemetry::Instrumentation::Base`
+
+The entry point of your instrumentation should be implemented as a subclass of `OpenTelemetry::Instrumentation::Base`:
+
+* Implement `install` block, where all of the integration work happens
+* Implement `present` block, which is you check if the library you are instrumenting was loaded
+* Implement `compatible` block and check for at least the minumum required library version
+* Any custom `options` you want to support
+
+The example below demonstrates how to implement the `Werewolf` instrumentation:
+
+```ruby
+
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Werewolf
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('0.1.0')
+
+        install do |_config|
+          require_relative 'handlers'
+          Handlers.subscribe
+        end
+
+        option :transformations, default: :omit, validate: %i[omit include]
+
+        present do
+          defined?(::Werewolf) && defined?(::ActiveSupport)
+        end
+
+        compatible do
+          Gem::Version.new(::Wereworlf::VERSION) >= MINIMUM_VERSION
+        end
+      end
+    end
+  end
+end
+
+```
+
+* The `install` block lazily requires the instrumentation handlers, which subscribe to events published by the `Werewolf` event hooks.
+* The `present` block checks if the `Werewolf` and `ActiveSupport` libraries are loaded, which it will use to subscribe to events and generate spans. It will skip the installation if those dependencies were not loaded before the instrumentation was being initialized.
```

Comment:
```suggestion
* The `present` block checks if the `Werewolf` and `ActiveSupport` libraries are loaded, which it will use to subscribe to events and generate spans. It will skip the installation if those dependencies were not loaded before the instrumentation was initialized.
```